pkg/machines/components/deleteDialog.jsx

```diff
@@ -103,16 +115,19 @@ export class DeleteDialog extends React.Component {
         Object.keys(vm.disks).sort()
                 .forEach(t => {
                     let d = vm.disks[t];
-                    if (d.type == 'file' && d.source.file)
-                        disks.push({ target: d.target, file: d.source.file, checked: !d.readonly });
+                    let entry = d;
```

Comment:
This is just an alias for the same object, so the line below will actually modify the original `vm.disks` array. This is confusing at best (use `d` directly then), but does this require setState() here?

Reply:
Oh good catch. I fixed it to use object.assign on the d object.
It needs sets state, because the this.state.disks object is the one that we parse to show the disks that are preselected on not in the delete dialog. It's part of the components state.

Reply:
Sorry, my question about setState() was very confusing. I meant it relative to directly modifying `vm.disks[t]`. It looks right now.

BTW, it's a matter of  preference, but I find the spread operator quite convenient and readable for these cases: `{...d, checked: !d.readonly}`